user/Makefile

```diff
@@ -2,20 +2,34 @@
 
 include ../Makefile.common
 
-SUBDIRS = $(shell find -mindepth 1 -maxdepth 1 -type d -not -name sysroot)
+PROGDIRS = $(shell find -mindepth 1 -maxdepth 1 -type d)
+PROGDIRS-inst = $(PROGDIRS:%=%.inst)
 
-.PHONY: clean sysroot all $(SUBDIRS)
+.PHONY: clean all $(PROGDIRS)
 
-all: $(SUBDIRS)
+all: $(PROGDIRS) install
 
-$(SUBDIRS): sysroot
+install: $(PROGDIRS-inst)
+
+$(PROGDIRS):
+	@echo "Maybe compiling $@..."
 	$(MAKE) -C $@
 
-sysroot: sysroot/lib/libmimiker.a
-sysroot/lib/libmimiker.a: start.o syscalls.o
+# .inst targets represent the installation of a program into sysroot.
+%.inst: %
```

Comment:
From my experience (`gcc`, `binutils` and generally speaking GNU programs source code) such targets are called `%-install`. It would be nice to find reasonable document describing Makefile conventions ([GNU](https://www.gnu.org/prep/standards/html_node/Makefile-Conventions.html) is not the best example) and stick to it.